pkg/kubelet/kubelet_node_status.go

```diff
@@ -739,12 +742,19 @@ func (kl *Kubelet) defaultNodeStatusFuncs() []func(context.Context, *v1.Node) er
 	// Volume limits
 	setters = append(setters, nodestatus.VolumeLimits(kl.volumePluginMgr.ListVolumePluginWithLimits))
 
+	var readyFuncs = []func() error{
+		kl.runtimeState.runtimeErrors,
+		kl.runtimeState.networkErrors,
+		kl.runtimeState.storageErrors,
+		kl.shutdownManager.ShutdownStatus,
+		validateWritableDirectory("root", kl.getRootDir()),
+		validateWritableDirectory("pods", kl.getPodsDir()),
+	}
 	setters = append(setters,
 		nodestatus.MemoryPressureCondition(kl.clock.Now, kl.evictionManager.IsUnderMemoryPressure, kl.recordNodeStatusEvent),
 		nodestatus.DiskPressureCondition(kl.clock.Now, kl.evictionManager.IsUnderDiskPressure, kl.recordNodeStatusEvent),
 		nodestatus.PIDPressureCondition(kl.clock.Now, kl.evictionManager.IsUnderPIDPressure, kl.recordNodeStatusEvent),
-		nodestatus.ReadyCondition(kl.clock.Now, kl.runtimeState.runtimeErrors, kl.runtimeState.networkErrors, kl.runtimeState.storageErrors,
-			validateHostFunc, kl.containerManager.Status, kl.shutdownManager.ShutdownStatus, kl.recordNodeStatusEvent, kl.supportLocalStorageCapacityIsolation()),
+		nodestatus.ReadyCondition(kl.clock.Now, readyFuncs, validateHostFunc, kl.containerManager.Status, kl.recordNodeStatusEvent, kl.supportLocalStorageCapacityIsolation()),
```

Comment:
I think updating the node condition to NotReady will start evicting the existing pod on the node. Is this what we want to achieve as well besides make the node unschedule when the filesystem  become readonly ? If not, probably could add the unscheduable taints to the node instead of setting the status as NotReady.

Reply:
The other readiness checks do the same.  The node lifecycle controller will add a taint to the node when it sees the node ready condition go to false. There's no where that I'm aware of in kubelet that manually marks the node as unschedulable, it's a user action.